src/main/java/app/ProgramController.java

```diff
@@ -0,0 +1,39 @@
+package app;
+
+import app.Program;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import app.ProgramRepository;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Controller
+public class ProgramController {
+
+    @Autowired
+    private ProgramRepository program;
+
+    @RequestMapping("/listPrograms")
+    public String listPrograms(Model model){
+        model.addAttribute("programs", program.findAll());
+        return "listPrograms";
+    }
+
+    @GetMapping("/pickProgramAndYear")
+    public String pickProgramAndYearForm(Model model){
+        int[] yearsArr = {2015, 2016, 2017, 2018};
+        ArrayList years = new ArrayList();
```

Comment:
I don't think years should be added this way. This is basically hardcoding it in the controller. An alternative would be to create an enum class and and assign a few years there and access them here.

Reply:
I agree. I was thinking it could just take the last 4 years by default - not hardcoded of course. In the later iterations it would make more sense to only include years for which we have courses.